hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java

```diff
@@ -146,8 +147,15 @@ public void preHandle(@NonNull final PreHandleContext context) throws PreCheckEx
         }
 
         // If the fee schedule key is changed then the transaction must also be signed by the new fee schedule key
-        if (op.hasFeeScheduleKey()) {
-            context.requireKeyOrThrow(op.feeScheduleKey(), INVALID_FEE_SCHEDULE_KEY);
+        if (op.hasFeeScheduleKey()
+                && !KeyOneOfType.UNSET.equals(op.feeScheduleKey().key().kind())) {
```

Comment:
I don't think you need this checking: !KeyOneOfType.UNSET.equals(op.feeScheduleKey().key().kind())). requireKeyOrThrow() takes care of it.

Reply:
You can see for example `updateTheFeeScheduleKeyToEmpty` test. It uses `withEmptyFeeScheduleKey` that sets an empty Key. When we send it like this the `op.hasFeeScheduleKey()` is **true** and we try to verify the key but it doesn't exist. We need this additional check so we know that if there is a key it's not UNSET